src/components/todosItem.tsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+
+import { useContext } from 'react';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './todosContext';
+import { FilterContext } from './filterContext';
+import { ManageCheckboxContext } from './manageCheckboxContext';
+import classNames from 'classnames';
+
+/* eslint-disable jsx-a11y/control-has-associated-label */
+interface ItemProps {
+  item: Todo;
+}
+
+export const TodosItem: React.FC<ItemProps> = ({ item }) => {
+  const { todos, setTodos, handleDeleteTodo } = useContext(TodosContext);
+  const { setIsChecked } = useContext(ManageCheckboxContext);
+  const { allId } = useContext(TodosContext);
+  const { isSelected } = useContext(FilterContext);
+
+  if (todos.every(element => element.completed === true)) {
+    setIsChecked(true);
+  } else {
+    setIsChecked(false);
+  }
+
+  const changePersonalComplete = () => {
+    return todos.map(elem => {
+      if (elem.id === item.id) {
+        return {
+          ...elem,
+          completed: !item.completed,
+        };
+      }
+
+      return elem;
+    });
+  };
+
+  enum FilterStatuses {
+    All = 'All',
+    Active = 'Active',
+    Completed = 'Completed',
+  }
+
+  const toRender = () => {
+    if (isSelected === FilterStatuses.All) {
+      return true;
+    } else if (isSelected === FilterStatuses.Active && !item.completed) {
+      return true;
+    } else if (
+      isSelected === FilterStatuses.Completed &&
+      item.completed === true
```

Comment:
```suggestion
      item.completed
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**